apps/web/src/pages/activities/ActivitiesPage.tsx

```diff
@@ -8,43 +8,64 @@ import { getActivityList } from '../../api/activity';
 import PageContainer from '../../components/layout/components/PageContainer';
 import PageMeta from '../../components/layout/components/PageMeta';
 import PageHeader from '../../components/layout/components/PageHeader';
-import { Select, Input } from '../../design-system';
+import { Select, Input, Button } from '../../design-system';
 import { ActivityStatistics } from './components/ActivityStatistics';
 import { ActivityGraph } from './components/ActivityGraph';
 import { ActivityList } from './components/ActivityList';
 import { ExecutionDetailsModal } from '../../components/activity/ExecutionDetailsModal';
+import { IActivityGraphStats } from './interfaces';
+import { useDebounce } from '../../hooks/useDebounce';
 
 interface IFiltersForm {
-  channels?: ChannelTypeEnum[];
+  channels: ChannelTypeEnum[];
+  templates: string[];
+  transactionId: string;
+  search: string;
 }
 
+const initialFormState: IFiltersForm = {
+  channels: [],
+  templates: [],
+  transactionId: '',
+  search: '',
+};
+
 export function ActivitiesPage() {
   const { templates, loading: loadingTemplates } = useTemplates(0, 100);
   const [page, setPage] = useState<number>(0);
   const [isModalOpen, setToggleModal] = useState<boolean>(false);
   const [notificationId, setNotificationId] = useState<string>('');
-  const [filters, setFilters] = useState<IFiltersForm>({ channels: [] });
+  const [filters, setFilters] = useState<IFiltersForm>(initialFormState);
   const { data, isLoading, isFetching } = useQuery<{ data: any[]; totalCount: number; pageSize: number }>(
     ['activitiesList', page, filters],
     () => getActivityList(page, filters),
     { keepPreviousData: true }
   );
 
-  function onFiltersChange(formData: IFiltersForm) {
-    setFilters(formData);
+  function onFiltersChange(formData: Partial<IFiltersForm>) {
+    setFilters((old) => ({ ...old, ...formData }));
   }
 
+  const debouncedTransactionIdChange = useDebounce((transactionId: string) => {
+    onFiltersChange({ transactionId });
+  }, 500);
+
+  const debouncedSearchChange = useDebounce((search: string) => {
+    onFiltersChange({ search });
+  }, 500);
```

Comment:
debounce the `onFiltersChange` call for the search/transaction id field changes... like when they change we don't want to set the filters right away because it's causing the request to the server, we want to delay it a little